UI/script.js

```diff
@@ -1,561 +1,852 @@
+      editedItem.photoLink = item.photoLink;
+    }
 
-      return true;
+    if (Object.prototype.hasOwnProperty.call(item, 'hashTags')) {
+      editedItem.hashTags = item.hashTags;
     }
 
-    add(item) {
-      if (!AdCollection.validate(item)) {
-        return false;
-      }
-      let tempItem = this._adList.find((a) => a.id === item.id);
-      if (tempItem) {
-        console.log('The array already has an element with this id');
-        return false;
-      }
-      this._adList.push(item);
+    if (Object.prototype.hasOwnProperty.call(item, 'discount')) {
+      editedItem.discount = item.discount;
+    }
 
-      return true;
+    if (Object.prototype.hasOwnProperty.call(item, 'validUntil')) {
+      editedItem.validUntil = item.validUntil;
     }
 
-    edit(id, item) {
-      let previous = this._adList.find((a) => a.id === id);
-      if (!previous) {
-        console.log('Element not found!');
-        return false;
-      }
+    if (Object.prototype.hasOwnProperty.call(item, 'rating')) {
+      editedItem.rating = item.rating;
+    }
 
-      if (Object.prototype.hasOwnProperty.call(item, 'id')) {
-        if (previous.id !== item.id) {
-          console.log("You mustn't change id!");
-          return false;
-        }
-      }
+    if (Object.prototype.hasOwnProperty.call(item, 'reviews')) {
+      editedItem.reviews = item.reviews;
+    }
+    if (!Model.validate(editedItem)) {
+      return false;
+    }
 
-      if (Object.prototype.hasOwnProperty.call(item, 'vendor')) {
-        if (previous.vendor !== item.vendor) {
-          console.log("You mustn't change vendor!");
-          return false;
-        }
-      }
+    this._ads.splice(this._ads.indexOf(previous), 1, editedItem);
+    return true;
+  }
+}
+
+class View {
+  constructor(_ads, model) {
+    this.model = model;
+    this._currentUser = null;
+    this._sortType = 'Date';
+    this._adsNumber = 10;
+  }
 
-      if (Object.prototype.hasOwnProperty.call(item, 'createdAt')) {
-        if (previous.createdAt.getTime() !== item.createdAt.getTime()) {
-          console.log("You mustn't change createdAt data!");
-          return false;
-        }
-      }
-      let editedItem = Object.assign(previous, item);
+  setSortType(sortType) {
+    this._sortType = sortType;
+    this.init();
+  }
 
-      if (Object.prototype.hasOwnProperty.call(item, 'description')) {
-        editedItem.description = item.description;
-      }
+  _updateButtonsUnderAd(ad) {
+    if (!this._currentUser) {
+      ad.querySelector('.leave-review').hidden = true;
+    }
+    if (ad.querySelector('.vendor').textContent !== `Поставщик: ${this._currentUser}`) {
+      ad.querySelector('.delete').hidden = true;
+      ad.querySelector('.edit').hidden = true;
+    }
+  }
 
-      if (Object.prototype.hasOwnProperty.call(item, 'link')) {
-        editedItem.link = item.link;
-      }
+  _updateGeneralButtons() {
+    const MESSAGE = document.getElementById('name');
```

Comment:
Capitalized words are most commonly used for shared between methods constants, or some constant literals, here we may write `message` (the same applies all consts).